code/modules/mob/living/carbon/diona_base.dm

```diff
@@ -515,6 +521,46 @@ var/list/diona_banned_languages = list(
 		else
 			to_chat(src, "<span class='danger'>You have forgotten the [L.name] language!</span>")
 
+/mob/living/carbon/alien/diona/proc/switch_to_gestalt()
+	set name = "Switch to Gestalt"
+	set desc = "Allows you to switch control back to your parent Gestalt."
+	set category = "Abilities"
+
+	if(!gestalt)
+		to_chat(src, span("warning", "You have no Gestalt!"))
+	else if(gestalt.stat == DEAD)
+		to_chat(src, span("danger", "Your Gestlat is not responding! Something could have happened to it!"))
+	else
+		gestalt.key = key
+
+/mob/living/carbon/alien/diona/proc/merge_back_to_gestalt()
+	set name = "Merge to Gestalt"
+	set desc = "Allows you to merge back to your parent Gestalt."
+	set category = "Abilities"
+
+	for(var/mob/living/carbon/human/diona/C in view(src, 7))
+		if(C == gestalt)
+			C.nutrition += REGROW_FOOD_REQ
+			C.DS.stored_energy += REGROW_ENERGY_REQ
+			total_radiation += 5
+			C.key = src.key
+			if(C.active_timers)
+				for(var/v in C.active_timers)
```

Comment:
Yeah, but that's way more code than iteration over list of timers. Especially because limbs can regenerate and then we need to update the timer ref to another timer.